### DISCUSSION TITLE
how to get value from rich text edit

### DISCUSSION THREAD

**Discussion by charis16** [mantinedev/mantine#3286] [Q&A] [answered]
I have a formList like:
`  const form = useForm({
    initialValues: {
      name: '',
      description: '',
      period: '',
      threshold: undefined,
      charge: undefined,
    } as PlanPackageFormType,
  });`

Now, I want to get values from rich text editor, how can i achieve that ? 
here my rich text editor i already put but value is still not set
 
`<RichTextEditor editor={editor} { ...form.getInputProps('description')}>
                <RichTextEditor.Toolbar sticky stickyOffset={60}>
                  <RichTextEditor.ControlsGroup>
                    <RichTextEditor.Bold />
                    <RichTextEditor.Italic />
                    <RichTextEditor.Underline />
                    <RichTextEditor.Strikethrough />
                    <RichTextEditor.ClearFormatting />
                    <RichTextEditor.Highlight />
                    <RichTextEditor.Code />
                  </RichTextEditor.ControlsGroup>

                  <RichTextEditor.ControlsGroup>
                    <RichTextEditor.H1 />
                    <RichTextEditor.H2 />
                    <RichTextEditor.H3 />
                    <RichTextEditor.H4 />
                  </RichTextEditor.ControlsGroup>

                  <RichTextEditor.ControlsGroup>
                    <RichTextEditor.AlignLeft />
                    <RichTextEditor.AlignCenter />
                    <RichTextEditor.AlignJustify />
                    <RichTextEditor.AlignRight />
                  </RichTextEditor.ControlsGroup>
                </RichTextEditor.Toolbar>

                <RichTextEditor.Content />
              </RichTextEditor>`

**Comment by AxelReid** [comment, marked as answer]
Thi is currently how i'm doing:
```js
const editor = useEditor({
    ...
    onUpdate(props) {
      const val = props.editor.getHTML()
    },
  })
```
But i need it to work with the form.
Anyone, let me know if u have a soultion

**Reply by shawnmclean** [reply]
@AxelReid figured this out yet?

**Comment by disillusionedowl** [comment]
As per https://tiptap.dev/api/editor#content I think this is how you set initial content from form defaults:

```js
const editor = useEditor( {
  ...
  content: form.getInputProps('description').value
  ...
} );
```
I don' think spreading the entirety of `...form.getInputProps('description')` will work on `RichTextEditor` since it is not a classic form element like text input or textarea for which that was intended. You could also try adding a hidden textarea to connect the two worlds somehow.

There is also the option to hook into the `RichTextEditor` context from anywhere https://mantine.dev/others/tiptap/#editor-context

Or create `Input.Wrapper` around the editor to mimick a classic form element, as per: https://github.com/mantinedev/mantine/discussions/3685#discussioncomment-5208932

**Reply by shawnmclean** [reply]
I'm still not following to be able to put the full thing together, are you able to give a full POC?

My code is open here if you want to take a look: https://github.com/samelogic/main/blob/master/apps/web/components/writings/compose/Editor/Editor.tsx

**Reply by disillusionedowl** [reply]
If you would like someone to take a closer look, your best chance is to create a reduced case at codesandbox.io and share it here.

**Reply by shawnmclean** [reply]
The basic looks like this:

```
export type FormProps = {
  title?: string;
  body?: object; // tiptap json content
};

const Editor = ({ title, body }: FormProps) => {
  const form = useForm<FormProps>({
    initialValues: { title, body },
  });
  const editor = useEditor({
    extensions: [StarterKit],
    onUpdate: (props) => {
      form.setValues({ body: props.editor.getJSON() });
    },
    content: form.getInputProps('body').value,
  });

  return (
    <form onSubmit={form.onSubmit((values) => console.log(values))}>
      <RichTextEditor editor={editor}>
        <RichTextEditor.Toolbar />
        <TextInput
          placeholder="Enter title..."
          variant="unstyled"
          size="xl"
          {...form.getInputProps("title")}
        />
        <Input.Wrapper {...form.getInputProps("body")}>
          <RichTextEditor.Content />
        </Input.Wrapper>
      </RichTextEditor>
    </form>
  );
};
```

**Reply by disillusionedowl** [reply]
I tried that code above without submit and it logs fine (even without having that Input.Wrapper, maybe it's not needed after all or maybe only for onSubmit and not onChange):

`<form onChange={console.log( form.values.body )} >`

so maybe something wrong with your submit handler?
